app/components/move_message.py

```diff
@@ -663,171 +630,283 @@ async def discard(
+                ),
+            ),
+            view=ContinueEditing(self, thread),
         )
-        return
+        await asyncio.sleep(remaining.total_seconds())
+        if thread.id not in self.edit_threads:
+            # The user finished or canceled editing of the thread while we were asleep.
+            return
+        # We once again need to re-calculate how much of the timeout has been elapsed
+        # to account for the last thread update time. This is especially important this
+        # time around since it's how the "Continue Editing" button signals to us that
+        # the user opted to continue.
+        elapsed = dt.datetime.now(tz=dt.UTC) - self.edit_threads[thread.id].last_update
+        if elapsed >= dt.timedelta(minutes=15):
+            # Fifteen minutes (time_to_warning's delta + remaining) have passed, so we
+            # shall delete the thread now.
+            await self.remove_edit_thread(thread, author, action="abandoned editing of")
+            return
+        # Restart the wait time. The start of this function will deal with the remaining
+        # time left to sleep.
+        await self.remove_edit_thread_after_timeout(thread, author)
+
+    @dc.app_commands.default_permissions(manage_messages=True)
+    @dc.app_commands.guild_only()
+    async def move_message(
+        self, interaction: dc.Interaction, message: dc.Message
+    ) -> None:
+        """
+        Adds a context menu item to a message to move it to a different channel. This is
+        used as a moderation tool to make discussion on-topic.
+        """
+        assert not is_dm(interaction.user)
+
+        if not (is_mod(interaction.user) or is_helper(interaction.user)):
+            await interaction.response.send_message(
+                "You do not have permission to move messages.", ephemeral=True
+            )
+            return
+
+        if not message_can_be_moved(message):
+            await interaction.response.send_message(
+                "System messages cannot be moved.",
+                ephemeral=True,
+                view=DeleteInstead(message),
+            )
+            return
 
-    if not message_can_be_moved(message):
         await interaction.response.send_message(
-            f"System messages cannot be turned into <#{config.help_channel_id}> posts.",
+            "Select a channel to move this message to.",
+            view=SelectChannel(self.bot, message, executor=interaction.user),
             ephemeral=True,
-            view=DeleteInstead(message),
         )
-        return
 
-    await interaction.response.send_modal(HelpPostTitle(message))
+    @dc.app_commands.default_permissions(manage_messages=True)
+    @dc.app_commands.guild_only()
+    async def turn_into_help_post(
+        self, interaction: dc.Interaction, message: dc.Message
+    ) -> None:
+        """
+        An extension of the move_message function that creates a #help post and then
+        moves the message to that channel.
```

Comment:
I know you didn't write this but would you mind updating it to not refer to functions while you're at it?

```suggestion
        An extension of "move message" that creates a #help post and then moves the
        message to that channel.
```